iceberg/iceberg-handler/src/main/java/org/apache/iceberg/mr/hive/HiveIcebergFilterFactory.java

```diff
@@ -97,22 +109,36 @@ private static Expression translate(ExpressionTree tree, List<PredicateLeaf> lea
    * @return Expression fully translated from Hive PredicateLeaf
    */
   private static Expression translateLeaf(PredicateLeaf leaf) {
-    String column = leaf.getColumnName();
+    TransformSpec transformSpec = TransformSpec.fromStringWithColumnName(leaf.getColumnName());
+    String column = transformSpec.getColumnName();
+    UnboundTerm<Object> columnTransform = getPartitionTransform(column, transformSpec);
+    boolean withoutTransform = columnTransform == null;
+
     switch (leaf.getOperator()) {
       case EQUALS:
-        Object literal = leafToLiteral(leaf);
-        return NaNUtil.isNaN(literal) ? isNaN(column) : equal(column, literal);
+        Object literal = leafToLiteral(leaf, transformSpec);
+        return NaNUtil.isNaN(literal) ?
+            (withoutTransform ? isNaN(column) : isNaN(columnTransform)) :
```

Comment:
could we substitute `column` with `columnTransform` and don't bring a ternary operator inside of each case block?

Reply:
done